hs-bindgen/src-internal/HsBindgen/Artefact.hs

```diff
@@ -140,6 +168,21 @@ runArtefacts
       FinalModuleSafe     -> liftIO backendFinalModuleSafe
       FinalModuleUnsafe   -> liftIO backendFinalModuleUnsafe
       FinalModules        -> liftIO backendFinalModules
+      -- File writes.
+      PendingFileWrite what path content -> do
+        fileExists <- liftIO $ doesFileExist path
+        case (fop, content) of
+          (ProtectExistingFiles, _)
+            | fileExists ->
+              throwError (FileAlreadyExistsException path)
+          (_, TextContent c) ->
```

Comment:
I am unsure if this is having the desired effect: Write all files, or none.

The `PendinfFileWrite` artefacts will still be executed one after the other, and if one fails in the middle, the earlier ones will have happened already. Is that right, or am I missing something?

I think we could have a "file system operation" storage in the artefact monad. Only at the end of the execution, when no Error trace has happened, these file system operations are executed one after the other.

Reply:
Yes, I think you are right and I misunderstood what the code was doing!

Reply:
Yes, I think you are right and I misunderstood what the code was doing!